calc/l3_route_resolver.go

```diff
@@ -688,17 +709,54 @@ func (r *RouteTrie) RemoveHost(cidr ip.V4CIDR, nodeName string) {
 	})
 }
 
-func (r *RouteTrie) AddLocalWEP(cidr ip.V4CIDR) {
+func (r *RouteTrie) AddWEP(cidr ip.V4CIDR, nodename string) {
 	r.updateCIDR(cidr, func(ri *RouteInfo) {
-		ri.LocalWEP.RefCount++
+		if ri.WEP.RefCount == nil {
+			ri.WEP.RefCount = map[string]int{}
+		}
+		ri.WEP.RefCount[nodename]++
+
+		// Groom the nodename slice.
+		ri.WEP.NodeNames = []string{}
+		for nodename, _ := range ri.WEP.RefCount {
+			ri.WEP.NodeNames = append(ri.WEP.NodeNames, nodename)
+		}
+		sort.Strings(ri.WEP.NodeNames)
 	})
 }
 
-func (r *RouteTrie) RemoveLocalWEP(cidr ip.V4CIDR) {
+func (r *RouteTrie) RemoveWEP(cidr ip.V4CIDR, nodename string) {
+	removeElement := func(s []string, e string) {
+		// Find element index
+		var i int
+		var v string
+		var found bool
+		for i, v = range s {
+			if v == e {
+				found = true
+				break
+			}
+		}
+		if !found {
+			log.Fatalf("Unable to find element %s in %s", e, s)
+		}
+
+		// Remove it.
+		s = append(s[:i], s[i+1:]...)
```

Comment:
Linter spotted that this side effect is not published.  When you do 
```
s = append(s[:i], s[i+1:]...)
```
then append will re-use the same backing array but the change in length of the slice is local to the new `s` that is returned, the input `s` will still have the old length.